### PR TITLE
Annotation edit tool (pt 1)

### PR DESCRIPTION
**Pull request by nyalldawson**
This PR adds:

- A GUI registry for annotation layer items, modeled heavily off the approach used by layout items in order to provide a nice clean core/gui separation between annotation item classes. Currently this registry handles creation of widgets for modifying annotation items of different types
- An initial start on a map tool for modifying annotation items. This isn't (yet) shown in the QGIS UI, but the tools currently allows for users to select an existing annotation item and modify its properties (symbol styling) via the layer styling dock. 

https://user-images.githubusercontent.com/1829991/131968778-0d4a1ef5-548a-4a27-a29f-7f02becff871.mp4

The tool can be tested through the python console:

```
iface.mapCanvas().setMapTool(iface.modifyAnnotationTool())
```
(works best with https://github.com/qgis/QGIS/pull/44919 included too) 

Follow up PRs will allow the map tool to modify the shape of annotation items and move them around.

### CODE REVIEW

File: tests/src/app/testqgsmaptooleditannotation.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+  QSignalSpy spy( &tool, &QgsMapToolModifyAnnotation::itemSelected );
+  TestQgsMapToolUtils utils( &tool );
+
+  // click outside of items
+  utils.mouseMove( 9, 9 );
+  utils.mouseClick( 9, 9, Qt::LeftButton, Qt::KeyboardModifiers(), true );
+  QCOMPARE( spy.count(), 0 );
+
+  // click on items
```
Comment (benoitdm-oslandia) [inline tests/src/app/testqgsmaptooleditannotation.cpp:122]: can you add a test for multiple object selection?

Comment (nyalldawson) [reply tests/src/app/testqgsmaptooleditannotation.cpp:122]: That's not currently supported - only single item selection is

File: src/app/annotations/qgsannotationitempropertieswidget.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  l->setContentsMargins( 0, 0, 0, 0 );
+  l->addWidget( mStack );
+  setLayout( l );
+
+  syncToLayer( layer );
+}
+
+void QgsAnnotationItemPropertiesWidget::syncToLayer( QgsMapLayer *layer )
+{
+  if ( layer == mLayer )
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/annotations/qgsannotationitempropertieswidget.cpp:45]: Done

**Comment by nyalldawson** [comment]
Unrelated test failure

**Pull request merged**